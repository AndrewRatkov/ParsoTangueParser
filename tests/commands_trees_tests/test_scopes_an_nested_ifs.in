int x := 30;
if x then
    int y := x + 1;
    if (y > 30) then
        str z := "OK";
    else
        str z := "BAD";
    fi
else
    int z := 999;
    if (x + z) % 3 != 0 then
        str y := "BAD";
    else
        str y := "OK";
    fi
fi
int z := 1;
int y := z;
