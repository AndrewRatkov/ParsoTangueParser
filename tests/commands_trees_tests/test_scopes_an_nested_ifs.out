[DEF int x]--->[INT,30]
IF
|
+--->[INT,x]
|
THEN
|
+--->[DEF int y]--->[INT,+ ]------->[INT,x]
|                       |
|                       +---------->[INT,1]
|
+--->IF
|    |
|    +--->[INT,> ]------->[INT,y]
|    |        |
|    |        +---------->[INT,30]
|    |
|    THEN
|    |
|    +--->[DEF str z]--->[STR,"OK"]
|    |
|    ELSE
|    |
|    +--->[DEF str z]--->[STR,"BAD"]
|    |
|    FI
|
ELSE
|
+--->[DEF int z]--->[INT,999]
|
+--->IF
|    |
|    +--->[INT,!=]------->[INT,% ]------->[INT,+ ]------->[INT,x]
|    |        |               |               |
|    |        |               |               +---------->[INT,z]
|    |        |               |
|    |        |               +---------->[INT,3]
|    |        |
|    |        +---------->[INT,0]
|    |
|    THEN
|    |
|    +--->[DEF str y]--->[STR,"BAD"]
|    |
|    ELSE
|    |
|    +--->[DEF str y]--->[STR,"OK"]
|    |
|    FI
|
FI
[DEF int z]--->[INT,1]
[DEF int y]--->[INT,z]
